onnx/defs/schema.cc

```diff
@@ -20,6 +20,48 @@ void RegisterSchema(OpSchema&& schema) {
   OpSchemaRegistry::OpSchemaRegisterOnce ONNX_UNUSED registration = schema;
 }
 
+void VerifyFunction(const OpSchema& schema, OpName_Domain_Version_Schema_Map& map) {
+  auto function_proto = schema.GetFunctionBody();
+  if (!function_proto)
+    return;
+ 
+  auto domain = schema.domain();
+  checker::CheckerContext ctx;
+  std::unordered_map<std::string, int> op_set;
+  auto version_range =
+      OpSchemaRegistry::DomainToVersionRange::Instance().Map().at(domain);
+  if (function_proto->since_version() > version_range.second ||
+      function_proto->since_version() < version_range.first) {
+    fail_check("Invalid function version in '", function_proto->name(), "'");
+  }
+  op_set.insert({domain, (int)function_proto->since_version()});
+  ctx.set_opset_imports(op_set);
+  ctx.set_is_main_graph(false);
+  
+  checker::TENSOR_TYPES_MAP tc_map;
```

Comment:
This type checking part is more complex. I suggest removing this from the current PR into a separate one. There are two challenges. One is that if the function inputs are allowed to have one of several different types, then the checking becomes more complex. The simplest way to do that is to check for every possible type assignment to all variables. Secondly, input.GetTypes() may return a type-variable name instead of a type.